tests/extra-source-files/FooTH.hs

```diff
@@ -0,0 +1,12 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module FooTH (embedFile) where
+
+import Language.Haskell.TH
+import Language.Haskell.TH.Syntax
+
+embedFile :: FilePath -> Q Exp
+embedFile path = do
+  str <- runIO (readFile path)
+  addDependentFile path
```

Comment:
A question: I guess `addDependentFile` is mostly only needed for reloading in the REPL, right?  Otherwise I believe Bazel would track the dependency on the input file automatically.

Reply:
Well I just added it because it was the right thing to do. It doesn't hurt in either case, does it?